dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/NativeHttpRestClient.java

```diff
@@ -0,0 +1,344 @@
+
+    public SearchResponse search(SearchRequest searchRequest, RequestOptions options) {
+
+        final HttpGetWithEntity httpGetWithEntity = new HttpGetWithEntity();
+
+        try {
+
+            httpGetWithEntity.setURI(new URI(url  + "/" + searchRequest.indices()[0] + "/_search"));
+
+            httpGetWithEntity.setEntity(createEntity(searchRequest.source(), REQUEST_BODY_CONTENT_TYPE));
+
+            final CloseableHttpResponse response = client.execute(httpGetWithEntity);
+
+            if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+                System.err.println("Method failed: " + response.getStatusLine());
+            }else{
+                return parseEntity(response.getEntity(), SearchResponse::fromXContent);
+            }
+
+        } catch (IOException e) {
+            System.err.println("Fatal transport error: " + e.getMessage());
+            e.printStackTrace();
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        } finally {
+            // Release the connection.
+            httpGetWithEntity.releaseConnection();
+        }
+
+        return null;
+    }
+
+    HttpEntity createEntity(ToXContent toXContent, XContentType xContentType) throws IOException {
+        return createEntity(toXContent, xContentType, ToXContent.EMPTY_PARAMS);
+    }
+
+    HttpEntity createEntity(ToXContent toXContent, XContentType xContentType, ToXContent.Params toXContentParams)
+            throws IOException {
+        BytesRef source = XContentHelper
+                .toXContent(toXContent, xContentType, toXContentParams, false).toBytesRef();
+        return new NByteArrayEntity(source.bytes, source.offset, source.length, createContentType(xContentType));
+    }
+
+    /**
+     * Ignores deprecation warnings. This is appropriate because it is only
+     * used to parse responses from Elasticsearch. Any deprecation warnings
+     * emitted there just mean that you are talking to an old version of
+     * Elasticsearch. There isn't anything you can do about the deprecation.
+     */
+    private static final DeprecationHandler DEPRECATION_HANDLER = new DeprecationHandler() {
+        @Override
+        public void usedDeprecatedName(String usedName, String modernName) {}
+        @Override
+        public void usedDeprecatedField(String usedName, String replacedWith) {}
+    };
+
+    protected final <Resp> Resp parseEntity(final HttpEntity entity,
+            final CheckedFunction<XContentParser, Resp, IOException> entityParser) throws IOException {
+        if (entity == null) {
+            throw new IllegalStateException("Response body expected but not returned");
+        }
+        if (entity.getContentType() == null) {
+            throw new IllegalStateException("Elasticsearch didn't return the [Content-Type] header, unable to parse response body");
+        }
+        XContentType xContentType = XContentType.fromMediaTypeOrFormat(entity.getContentType().getValue());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'xContentType' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5121287)